### ISSUE TITLE
Stacking dropout layers?

### ISSUE THREAD

**Issue by hsm207** [fouticus/pipgcn#1]
I just wanted to check with you about the design of some of the layers in nn_components.py.

In the nn_components,py file, the dense() function starts and ends with a call to the dropout layer ie:

    def dense(input, params, out_dims=None, dropout_keep_prob=1.0, nonlin=True, trainable=True, **kwargs):

       input = tf.nn.dropout(input, dropout_keep_prob)

       # some other code...

       Z = tf.nn.dropout(Z, dropout_keep_prob)

       return Z, params

This means that when this two layers are stacked together, then the output of the dense layer 1 (which already has dropout applied to it) will be passed through another dropout layer at the start of dense layer 2. A similar situation also occurs in no_conv layer.

Was this what was intended?

**Comment by fouticus** [comment]
Hi, good catch!

While simplifying the code for public release we discovered that we did in fact perform dropout twice on these layers. This was not intended, but we decided to leave it in to accurately reflect what was done for the paper. 

I hope this helps. 

Alex.
